src/main/java/com/salesforce/storm/spout/dynamic/retry/DefaultRetryManager.java

```diff
@@ -184,10 +184,11 @@ public MessageId nextFailedMessageToRetry() {
         }
 
         // Populate the values
-        final Long lowestTimestampKey = entry.getKey();
+        final long lowestTimestampKey = entry.getKey();
```

Comment:
This can generate a NPE can’t it?

Reply:
Most map implementations disallow null keys.  And we set the key values in the map to the result of Clock's current timestamp, so that would have to return null.

I'd say its a highly unlikely scenario.